cookbooks/aws-parallelcluster-environment/spec/unit/recipes/config_default_user_home_spec.rb

```diff
@@ -29,11 +29,13 @@
           expect(chef_run).to run_bash("Verify data integrity for #{user_home}").with(
             code: <<-CODE
     diff_output=$(diff -r #{user_home} #{user_local_home})
-    if [ $? -eq 0 ]; then
+    if [[ $diff_output != *"Only in #{user_home}"* ]]; then
```

Comment:
The integrity check should fail if there is at least one occurrence of the substring "Only in /tmp#{user_home}".
Here the check is doing the opposite: it fails if there is at least an occurrence of "Only in #{user_home}".

Reply:
No, I would disagree. "Only in #{user_home}" is okay in this use case. We sync data from /tmp#{user_home} to #{user_home}, and we want all data in /tmp#{user_home} have been sync to #{user_home}. But there maybe are some pre-existing files in #{user_home}. So we can accept "Only in #{user_home}". But if "Only in /tmp#{user_home}" appear, that means some data sync failed.

Reply:
And FYI, the #{user_home} in this case is not the original one. It has been replaced. So the /tmp#{user_home} contains the original data.